src/Fixer/Import/NoUnneededAliasFixer.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Import;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\NamespaceUseAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\NamespaceUsesAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class NoUnneededAliasFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Remove unneeded alias in `use` clauses.',
+            [new CodeSample("<?php\nnamespace Foo;\nuse Bar\\Baz as Baz;\n")]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run after the SingleImportPerStatementFixer (for fix separated use statements as well) and NoUnusedImportsFixer (just for save performance) and NoLeadingImportSlashFixer
```

Comment:
IMO this new fixer should be able to deal with grouped imports in case `single_import_per_statement` is not enabled. This would remove the priority issue. Same with `no_leading_import_slash`.

Reply:
Sadly NamespaceUsesAnalyzer skips group and multiple use

Reply:
Add support for multiple use (not group) with an extra argument to avoid side effect